From fee00c36f79dd06432392dd9d4891efc5d253fd2 Mon Sep 17 00:00:00 2001
From: Jean-Yves Avenard <jyavenard@mythtv.org>
Date: Tue, 3 Jun 2014 13:22:18 +1000
Subject: [PATCH 01/16] Remove hack forcing MPEG1 video to be treated as MPEG2

This hack was added to fix #287, which was a problem with XvMC decoding.
It makes no sense today, and actually break genuine MPEG1 videos.

Fixes #12160
---
 mythtv/libs/libmythtv/avformatdecoder.cpp |    8 --------
 1 file changed, 8 deletions(-)

diff --git a/mythtv/libs/libmythtv/avformatdecoder.cpp b/mythtv/libs/libmythtv/avformatdecoder.cpp
index b1cf3eb..c7f6a94 100644
--- a/mythtv/libs/libmythtv/avformatdecoder.cpp
+++ b/mythtv/libs/libmythtv/avformatdecoder.cpp
@@ -2374,14 +2374,6 @@ int AvFormatDecoder::ScanStreams(bool novideo)
                 bool foundgpudecoder = false;
 
 #ifdef USING_VDPAU
-                // HACK -- begin
-                // Force MPEG2 decoder on MPEG1 streams.
-                // Needed for broken transmitters which mark
-                // MPEG2 streams as MPEG1 streams, and should
-                // be harmless for unbroken ones.
-                if (enc->codec_id == AV_CODEC_ID_MPEG1VIDEO)
-                    enc->codec_id = AV_CODEC_ID_MPEG2VIDEO;
-                // HACK -- end
                 MythCodecID vdpau_mcid;
                 vdpau_mcid =
                     VideoOutputVDPAU::GetBestSupportedCodec(width, height, dec,
-- 
1.7.10.2

